core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -72,7 +73,7 @@ public PathMerger setDouglasPeucker(DouglasPeucker douglasPeucker) {
         return this;
     }
 
-    public PathMerger setPathDetailsBuilders(PathDetailsBuilderFactory pathBuilderFactory, List<String> requestedPathDetails) {
```

Comment:
request order might be significant (I know that json shouldn't be considered ordered but some clients might still do)

Reply:
The whole List vs Set boils down to how we should treat duplicates in the requested details. The current implementation doesn't really check for this case and will treat it the same way as a missing builder:

https://github.com/graphhopper/graphhopper/blob/598ecb0ff519a86112686e8a97adbfe9c3148aa7/core/src/main/java/com/graphhopper/util/details/PathDetailsBuilderFactory.java#L89-L91

Reply:
IMO this is not a bug as duplicating a path detail can be seen as an illegal argument.
Can we still request path details in a list with the POST endpoint? I'm also worried about different ordered in the output JSON causing trouble with clients.